src/main/java/lol/server/Server.java

```diff
@@ -41,81 +34,104 @@ public void run() {
-    players.add(new Player(socket, this));
-  }
-
-  private void startGame() throws IOException {
-    championSelectionPhase();
-    spawnPhase();
-    arena.startGamePhase();
-    ui.update();
-    wait(2000);
+    connectedUsers.add(new Player(socket, this));
+    playerScores.add(0);
   }
 
-  private void wait(int timeInMS) {
-    try {
-      Thread.sleep(timeInMS); // wait time in milliseconds to control duration
-    } catch(InterruptedException e) {
-      e.printStackTrace();
+ //tourney seeded into 2 groups depending on the sign up order.
+  private void tourneyOrganization() throws IOException {
+    ArrayList<Player> group1 = new ArrayList<Player>();
+    ArrayList<Player> group2 = new ArrayList<Player>();
+    for (int i = 0; i < connectedUsers.size(); i++) {
+      if(i % 2 == 0)
+        group1.add(connectedUsers.get(i));
+      else
+        group2.add(connectedUsers.get(i));
     }
-  }
-
-  private void endOfGameMessage() {
-    for(int i = 0; i < battlefield.numberOfTeams(); ++i) {
-      Nexus nexus = battlefield.nexusOf(i);
-      if(nexus.isAlive()) {
-        System.out.println("WINNER: " + nexus);
-      }
-      else {
-        System.out.println("LOSER: " + nexus);
-      }
+    if(group2.size() < group1.size()) {
+      group2.add(null); //add dummy player.
     }
-  }
-
-  private void gameLoop() throws IOException {
-    for(int turns = 0; battlefield.allNexusAlive() && turns < lol.client.Client.MAX_TURNS; ++turns) {
-      for(Player player : players) {
-        Turn turn = player.askTurn();
-        arena.applyTurn(turn);
-        ui.update();
-        broadcast(turn);
-        wait(2000);
+    //round robin rules according to wikipedia explanation.
+    //Dummy player round is skipped.
+    int set = 0;
+    int round = 0;
+    while(set < (int)((double)((group1.size()+group2.size())/2)*((group1.size()+group2.size())-1))) {
+      round = 0;
+      while (round < group1.size()) {
+        //group 2's dummy player is overgone.
+        if(group2.get(round) != null && group2.get(round) != null) {
+          roundOrganizer(group1.get(round), group2.get(round));
+        }
+        round++;
       }
+      set += round;
+      group1.add(1, group2.get(0));
+      group2.remove(0);
+      group2.add(group1.get(group1.size()-1));
+      group1.remove(group1.size()-1);
     }
+
   }
 
-  private void championSelectionPhase() throws IOException {
-    System.out.println("Champion selection phase...");
-    ArrayList<Turn> champSelect = new ArrayList<Turn>();
-    for(Player p : players) {
-      p.sendUID();
-      Turn turn = p.askTurn();
-      champSelect.add(turn);
-      arena.applyTurn(turn);
-    }
-    System.out.println("Broadcast team composition...");
-    broadcast(champSelect);
+  private void roundOrganizer(Player playerGroup1, Player playerGroup2) throws IOException {
+    Player bluePlayer = playerGroup1;
+    Player redPlayer = playerGroup2;
+    bluePlayer.setRoundUID(0);
```

Comment:
Use Nexus.BLUE/RED instead of 0/1 constant please (occur in several places in the code).
You can also use an array of players (of only two elements), to avoid repeating the code. For instance it would allow you to write `updateScores(players[winner])` instead of having a conditional.